### ISSUE TITLE
Latest stable version "2.0.26" is not working

### ISSUE THREAD

**Issue by ajaxe** [jonnyzzz/TeamCity.Node#191]
We upgraded Teamcity to latest version (156166) & our builds, that use this plugin, started to show `Incompatible runner:
Node.js Yarn` on the build agents. Looking into one of the agent logs I see the following error

```
[2024-04-02 xx:xx:xx,xxx]  ERROR - gins.spring.SpringPluginLoader - Error loading plugin 'teamcity-node-agent-2.0.26': Failed to initialize spring context: Error creating bean with name 'com.jonnyzzz.teamcity.plugins.node.agent.nvm.NVMRunner#0' defined in Byte array resource [plugin: teamcity-node-agent-2.0.26#teamcity-node-agent-2.0.26.jar!/META-INF/build-agent-plugin-node.xml]: Unsatisfied dependency expressed through constructor parameter 1; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'com.jonnyzzz.teamcity.plugins.node.agent.processes.CompositeProcessFactoryImpl#0': Lookup method resolution failed; nested exception is java.lang.IllegalStateException: Failed to introspect Class [com.jonnyzzz.teamcity.plugins.node.agent.processes.CompositeProcessFactoryImpl] from ClassLoader [PluginStandaloneClassLoader{plugin=teamcity-node-agent-2.0.26 parent=java.net.URLClassLoader@433c675d}]

org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'com.jonnyzzz.teamcity.plugins.node.agent.nvm.NVMRunner#0' defined in Byte array resource [plugin: teamcity-node-agent-2.0.26#teamcity-node-agent-2.0.26.jar!/META-INF/build-agent-plugin-node.xml]: Unsatisfied dependency expressed through constructor parameter 1; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'com.jonnyzzz.teamcity.plugins.node.agent.processes.CompositeProcessFactoryImpl#0': Lookup method resolution failed; nested exception is java.lang.IllegalStateException: Failed to introspect Class [com.jonnyzzz.teamcity.plugins.node.agent.processes.CompositeProcessFactoryImpl] from ClassLoader [PluginStandaloneClassLoader{plugin=teamcity-node-agent-2.0.26 parent=java.net.URLClassLoader@433c675d}]
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:800)
	at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:229)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1372)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1222)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:582)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:955)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:921)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:583)
	at jetbrains.buildServer.plugins.spring.SpringPluginLoader.pluginClassesLoaded(SpringPluginLoader.java:126)
	at jdk.internal.reflect.GeneratedMethodAccessor12.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at jetbrains.buildServer.util.EventDispatcher.invokeListeners(EventDispatcher.java:157)
	at jetbrains.buildServer.util.EventDispatcher.dispatch(EventDispatcher.java:136)
	at jetbrains.buildServer.util.EventDispatcher$2.invoke(EventDispatcher.java:82)
	at jdk.proxy2/jdk.proxy2.$Proxy22.pluginClassesLoaded(Unknown Source)
	at jetbrains.buildServer.plugins.PluginManagerImpl$5.visitPlugin(PluginManagerImpl.java:470)
	at jetbrains.buildServer.plugins.PluginsCollection$1.run(PluginsCollection.java:98)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'com.jonnyzzz.teamcity.plugins.node.agent.processes.CompositeProcessFactoryImpl#0': Lookup method resolution failed; nested exception is java.lang.IllegalStateException: Failed to introspect Class [com.jonnyzzz.teamcity.plugins.node.agent.processes.CompositeProcessFactoryImpl] from ClassLoader [PluginStandaloneClassLoader{plugin=teamcity-node-agent-2.0.26 parent=java.net.URLClassLoader@433c675d}]
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.determineCandidateConstructors(AutowiredAnnotationBeanPostProcessor.java:289)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.determineConstructorsFromBeanPostProcessors(AbstractAutowireCapableBeanFactory.java:1302)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1219)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:582)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1391)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1311)
	at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:887)
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:791)
	... 27 more
Caused by: java.lang.IllegalStateException: Failed to introspect Class [com.jonnyzzz.teamcity.plugins.node.agent.processes.CompositeProcessFactoryImpl] from ClassLoader [PluginStandaloneClassLoader{plugin=teamcity-node-agent-2.0.26 parent=java.net.URLClassLoader@433c675d}]
	at org.springframework.util.ReflectionUtils.getDeclaredMethods(ReflectionUtils.java:485)
	at org.springframework.util.ReflectionUtils.doWithLocalMethods(ReflectionUtils.java:321)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.determineCandidateConstructors(AutowiredAnnotationBeanPostProcessor.java:267)
	... 40 more
Caused by: java.lang.NoClassDefFoundError: jetbrains/buildServer/agent/BuildProcessFacade
	at java.base/java.lang.Class.getDeclaredMethods0(Native Method)
	at java.base/java.lang.Class.privateGetDeclaredMethods(Class.java:3402)
	at java.base/java.lang.Class.getDeclaredMethods(Class.java:2504)
	at org.springframework.util.ReflectionUtils.getDeclaredMethods(ReflectionUtils.java:467)
	... 42 more
Caused by: java.lang.ClassNotFoundException: Class 'jetbrains.buildServer.agent.BuildProcessFacade' was not found
	at jetbrains.buildServer.plugins.classLoaders.TeamCityClassLoader.doLoadClass(TeamCityClassLoader.java:92)
	at jetbrains.buildServer.plugins.classLoaders.TeamCityClassLoader.loadClass(TeamCityClassLoader.java:46)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	... 46 more
```

I would appreciate any resolution or manual workaround for this. Given recent security fixes for Teamcity, I would like to keep Teamcity downgrade as my last option.

**Comment by ajaxe** [comment]
Initially I did not take version 2.0.27 as it was not marked as "stable" on the Teamcity server. I am currently monitoring our builds.

I will post my update.

**Comment by ajaxe** [comment]
This is resolved by updating to version 2.0.27
